File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/FizzBuzz.java
package com.example.borysgurtovyi.codinginterviewprep;

import java.util.ArrayList;
import java.util.List;


/**
 * Created by borysgurtovyi on 1/29/18.
 */

public class FizzBuzz {

    public List<String> fizzBuzz(int n) {
        List<String> resultList = new ArrayList<>();
        for(int i = 1; i<=n; i++) {
            if(i%3 ==0 && i%5 ==0) {
               resultList.add("FizzBuzz");
            } else if (i%5==0) {
                resultList.add("Buzz");
            } else if(i%3 ==0) {
                resultList.add("Fizz");
            } else {
                resultList.add(i + "");
            }
        }
        return resultList;
    }

    // O(n)

}

File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/FizzBuzz.kt
package com.example.borysgurtovyi.codinginterviewprep

/**
 * Created by lisa on 06.02.2018.
 */

open class Solution{
    fun fizzBuzz(n: Int): MutableList<String> {
        var resultList: MutableList<String> = mutableListOf()
        for (i in 1..n) {
            if (i % 3 == 0 && i % 5 == 0) {
                resultList.add("FizzBuzz")
            } else if (i % 3 == 0) {
                resultList.add("Fizz")
            } else if (i % 5 == 0) {
                resultList.add("Buzz")
            } else resultList.add(i.toString())
        }
        return resultList
    }
}